k9mail/src/main/java/com/fsck/k9/controller/MessagingListener.java

```diff
@@ -13,173 +14,63 @@
+    void listFolders(Account account, List<LocalFolder> folders);
+    void listFoldersFinished(Account account);
+    void listFoldersFailed(Account account, String message);
 
-    public void pendingCommandsProcessing(Account account) {}
+    void listLocalMessagesAddMessages(Account account, String folder, List<LocalMessage> messages);
 
-    public void pendingCommandCompleted(Account account, String commandTitle) {}
+    void synchronizeMailboxStarted(Account account, String folder);
+    void synchronizeMailboxHeadersStarted(Account account, String folder);
+    void synchronizeMailboxHeadersProgress(Account account, String folder, int completed, int total);
+    void synchronizeMailboxHeadersFinished(Account account, String folder, int totalMessagesInMailbox,
+            int numNewMessages);
+    void synchronizeMailboxProgress(Account account, String folder, int completed, int total);
+    void synchronizeMailboxNewMessage(Account account, String folder, Message message);
+    void synchronizeMailboxRemovedMessage(Account account, String folder, Message message);
+    void synchronizeMailboxFinished(Account account, String folder, int totalMessagesInMailbox, int numNewMessages);
+    void synchronizeMailboxFailed(Account account, String folder, String message);
 
-    public void pendingCommandsFinished(Account account) {}
+    void loadMessageRemoteFinished(Account account, String folder, String uid);
+    void loadMessageRemoteFailed(Account account, String folder, String uid, Throwable t);
 
+    void checkMailStarted(Context context, Account account);
+    void checkMailFinished(Context context, Account account);
 
-    /**
-     * Called when a remote search is started
-     *
-     * @param folder
-     */
-    public void remoteSearchStarted(String folder) {}
+    void sendPendingMessagesStarted(Account account);
+    void sendPendingMessagesCompleted(Account account);
+    void sendPendingMessagesFailed(Account account);
 
+    void emptyTrashCompleted(Account account);
 
-    /**
-     * Called when server has responded to our query.  Messages have not yet been downloaded.
-     *
-     * @param numResults
-     */
-    public void remoteSearchServerQueryComplete(String folderName, int numResults, int maxResults) { }
+    void folderStatusChanged(Account account, String folderName, int unreadMessageCount);
+    void systemStatusChanged();
 
+    void messageDeleted(Account account, String folder, Message message);
+    void messageUidChanged(Account account, String folder, String oldUid, String newUid);
 
-    /**
-     * Called when a new result message is available for a remote search
-     * Can assume headers have been downloaded, but potentially not body.
-     * @param folder
-     * @param message
-     */
-    public void remoteSearchAddMessage(String folder, Message message, int numDone, int numTotal) { }
+    void setPushActive(Account account, String folderName, boolean enabled);
```

Comment:
This method name implies it's always setting push as active. But there's the enabled flag? Does it also notify when push is inactive?